drivers/kscan/Kconfig

```diff
@@ -27,4 +28,66 @@ config KSCAN_INIT_PRIORITY
 	help
 	  Keyboard scan device driver initialization priority.
 
+config KSCAN_MATRIX
+	bool
+	help
+	  A matrix circuit is used in most computer keyboards where the key
+	  switches are connected by a grid of wires, similar to a diode matrix.
+
+if KSCAN_MATRIX
+
+config KSCAN_MATRIX_MAX_COLUMNS
+	int "keyboard matrix max column"
+	default 18
+	help
+	  Maximum columns of the matrix circuit.
+
+config KSCAN_MATRIX_MAX_ROWS
+	int "keyboard matrix max row"
+	default 8
+	help
+	  Maximum rows of the matrix circuit.
```

Comment:
why isn't this part of Devicetree kscan devices? it could be included by kscan bindings, same it happens for width/height on displays.

Reply:
Nuvoton EC SoC can support maximum 18 columns x 8 rows for kscan matrix hardware interface.
However, in an application such as Chromebook, it usually uses 13X8 or 15x8 configuration and left the reminding pins for other usage. That's reason I added these two Kconfig entries. The application can overwrite the default values according to its requirement.